## Re-review verdict: **NOT yet ready to merge** — small CI fix required

### What was addressed since the last review ✅

- **Tests added for all three changed surfaces** — `TelegramSMSRenderedContentTestCase` (8 tests for TG/SMS/RCS rendered_content incl. None/empty placeholder_data), `TestTelegramBotAppCreate` (3 tests incl. invalid-token rejection + record cleanup + error-leak guard), `TestTelegramBroadcastCreate` (4 tests covering QUEUED w/ message, QUEUED w/o message, `text` alias, and DRAFT exemption). Coverage of the bug-fix surface area looks solid.
- **Docs updated** — `docs/src/content/docs/api/broadcast.mdx` now shows the correct `placeholder_data.message` shape with both SMS and Telegram (with media) examples plus a caution box on the QUEUED requirement. Matches the new contract.
- **Generic error message** — `telegram/serializers.py` and `telegram/viewsets/bot_app.py` no longer leak Telegram internals; new test `test_error_message_does_not_leak_telegram_internals` enforces it.

### Blocker 🔴 — Lint fails, tests are gated behind it

The `Lint & Security Scan` job is failing, which causes `Tests (Python 3.11/3.12)` and `Django System Checks` to be skipped (both have `needs: lint`). Two issues, both in `telegram/tests/test_viewset_guards.py`:

```
telegram/tests/test_viewset_guards.py:10:25: F401 [*] `django.urls.reverse` imported but unused
```

```
ruff format --check . →  Would reformat: telegram/tests/test_viewset_guards.py
```

Fix locally with:

```bash
ruff check --fix telegram/tests/test_viewset_guards.py
ruff format telegram/tests/test_viewset_guards.py
```

The format diff collapses one multi-line `get_or_create(...)` call and adds two blank lines before class definitions — purely cosmetic.

### Non-blocking follow-up (tracked from previous review)

`TelegramBroadcastViewSet.perform_create` only runs the content guard when `status == QUEUED` at create time. A `DRAFT` broadcast PATCH'd to `QUEUED` (or transitioned by the scheduler) still bypasses this guard — the `rendered_content` empty check is the safety net at send time. Worth a follow-up issue, not a blocker for this PR.

### Verdict

**Approve once green.** Code, tests, and docs are all in good shape. Please push the lint/format fix; once CI is green on all jobs, this is good to merge.

---
_Generated by [Claude Code](https://claude.ai/code)_

--- a/telegram/tests/test_viewset_guards.py
+++ b/telegram/tests/test_viewset_guards.py
@@ -7,7 +7,6 @@ from unittest.mock import patch
 
 import pytest
 from django.contrib.auth import get_user_model
-from django.urls import reverse
 from rest_framework.test import APIClient
 
 from broadcast.models import Broadcast, BroadcastPlatformChoices
@@ -35,9 +34,7 @@ def tenant(db, user):
 
 @pytest.fixture
 def role(tenant):
-    role, _ = TenantRole.objects.get_or_create(
-        tenant=tenant, slug="owner", defaults={"name": "Owner", "priority": 100}
-    )
+    role, _ = TenantRole.objects.get_or_create(tenant=tenant, slug="owner", defaults={"name": "Owner", "priority": 100})
     return role
 
 
@@ -54,6 +51,7 @@ def auth_client(user, tenant_user):
 
 
 # ── TelegramBotAppViewSet ────────────────────────────────────────────────────
+
 
 @pytest.mark.django_db
 class TestTelegramBotAppCreate:
@@ -107,6 +105,7 @@ class TestTelegramBotAppCreate:
 
 
 # ── TelegramBroadcastViewSet ─────────────────────────────────────────────────
+
 
 @pytest.mark.django_db
 class TestTelegramBroadcastCreate:

